### ISSUE TITLE
CSF n/a in confounds file

### ISSUE THREAD

**Issue by shanaahall** [poldracklab/fmriprep#2209]
### What version of fMRIPrep are you using?
20.1.0

### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?
Singularity

### What is the exact command-line you used?
```
qsub -l h_vmem=50G,v=50G -N sub-00727fmriprep -o /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/logs/FMRIprep_logs/Output-sub-00727 fmriprep-jobid-%j.out -b y "unset PYTHONPATH; singularity run -B /mnt -e /usr/local/packages/singularity/images/fmriprep-v20.1.0.biac.simg /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/data_BIDS /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/data_fmriprep participant --participant-label 00727 -w /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/fmriprep_workingdir --fs-license-file /mnt/BIAC/munin6/ConnNeuro.01/Scripts/license.txt --use-syn-sdc --use-aroma --error-on-aroma-warnings --output-spaces MNI152NLin2009cAsym MNI152NLin6Asym"
```

### Have you checked that your inputs are BIDS valid?
Yes

### Did fMRIPrep generate the visual report for this particular subject? If yes, could you share it?

Will emai

### Can you find some traces of the error reported in the visual report (at the bottom) or in *crashfiles*?
No errors were reported in the visual report, nor isthere a crash file that I can see.
```
NA
```

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?
Yes, using previously computed freesurfer results. One of the CSF files referenced in the log looks fine. I'll email that as well. 

### fMRIPrep log
If you have access to the output logged by fMRIPrep, please make sure to attach it as a text file to this issue.

Attached.
[Output-sub-00727fmriprep-jobid-%j.txt](https://github.com/poldracklab/fmriprep/files/4885754/Output-sub-00727fmriprep-jobid-.j.txt)

**Comment by mgxd** [comment]
hey @shanaahall - can you also share the troublesome confounds file?

**Comment by shanaahall** [comment]
Sure thing. They should be in the box folder that I sent a link to but I'll also just email you the csv files separately.

**Comment by shanaahall** [comment]
Hello. I was just following-up to see if there are any thoughts about this. 

Thank you!

**Comment by mgxd** [comment]
Sorry for the delay, I did not see anything in my inbox. Please resend any links to mathiasg(at)stanford.edu

**Comment by shanaahall** [comment]
Ok I have sent everything to your email!

**Comment by mgxd** [comment]
Ah, this is most likely related to what was discussed in https://github.com/poldracklab/fmriprep/issues/2155#issuecomment-645554923. 

There is a potential fix (https://github.com/poldracklab/fmriprep/pull/2130) in the works, targeting the next release. If you'd like to pilot the fix with your data:

1) Download the patch (https://github.com/oesteban/fmriprep/archive/fix/2129-compcor-masks-reliability.zip) and unzip
2) Replace the fmriprep in the container with the patch you downloaded
   - if you are using the docker wrapper, you can add the argument `--patch fmriprep=<path-to-fmriprep-patch>/fmriprep`
   - if you are using a docker command, add `-v <path-to-fmriprep-patch>/fmriprep:/usr/local/miniconda/lib/python3.7/site-packages/fmriprep:ro` to your bind mounts.
   - if you are using a singularity command, add `-B <path-to-fmriprep-patch>:/usr/local/miniconda/lib/python3.7/site-packages/fmriprep:ro` to your bind mounts.

**Comment by shanaahall** [comment]
Great! Thank you! I am using a singularity so I'm trying this now using this command: 

```qsub -l h_vmem=50G,v=50G -N sub-00727fmriprep -o /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/logs/FMRIprep_logs/Output-sub-00727 fmriprep-jobid-%j.out -b y "unset PYTHONPATH; singularity run -B /mnt/munin6/Meade/ConnNeuro.01/Scripts/Shana/fmriprep-fix-2129-compcor-masks-reliability:/usr/local/miniconda/lib/python3.7/site-packages/fmriprep:ro -e /usr/local/packages/singularity/images/fmriprep-v20.1.0.biac.simg /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/data_BIDS /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/data_fmriprep participant --participant-label 00727 -w /mnt/BIAC/munin6/ConnNeuro.01/Analysis/BIDS/fmriprep_workingdir --fs-license-file /mnt/BIAC/munin6/ConnNeuro.01/Scripts/license.txt --use-syn-sdc --use-aroma --error-on-aroma-warnings --output-spaces MNI152NLin2009cAsym MNI152NLin6Asym"```

and I'm getting an error: 
```Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 6, in <module>
    from fmriprep.cli.run import main
ModuleNotFoundError: No module named 'fmriprep.cli'

**Comment by mgxd** [comment]
@shanaahall you'll have to alter the binding to:

`-B /mnt/munin6/Meade/ConnNeuro.01/Scripts/Shana/fmriprep-fix-2129-compcor-masks-reliability/fmriprep:/usr/local/miniconda/lib/python3.7/site-packages/fmriprep:ro`

**Comment by shanaahall** [comment]
Alrighty I tried that and now I'm getting this error

```Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 10, in <module>
    sys.exit(main())
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/cli/run.py", line 17, in main
    parse_args()
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/cli/parser.py", line 565, in parse_args
    parser = _build_parser()
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/cli/parser.py", line 61, in _build_parser
    currentv = Version(config.environment.version)
  File "/usr/local/miniconda/lib/python3.7/site-packages/packaging/version.py", line 277, in __init__
    raise InvalidVersion("Invalid version: '{0}'".format(version))
packaging.version.InvalidVersion: Invalid version: 'fix/2129-compcor-masks-reliability'

**Comment by mgxd** [comment]
that's odd, previously we've been able to plug and play (see https://github.com/poldracklab/fmriprep/issues/2153#issuecomment-636224354)

@effigies @oesteban anything catching your eye?

**Comment by mgxd** [comment]
@shanaahall we pushed a docker image (`poldracklab/fmriprep:compcor-revision-2130`) to DockerHub with the patch already applied, hopefully that makes testing a bit easier.

**Comment by shanaahall** [comment]
It looks like it worked!!

**Comment by mgxd** [comment]
awesome to hear! we are still piloting the compcor revision, and hoping to include it in the next major release (20.2.0).

**Comment by oesteban** [comment]
Fixed with #2130
